trie/zk_merkle_state_trie.go

```diff
@@ -84,8 +87,16 @@ func (z *ZkMerkleStateTrie) UpdateContractCode(_ common.Address, _ common.Hash,
 	return nil
 }
 
-func (z *ZkMerkleStateTrie) DeleteStorage(_ common.Address, key []byte) error { return z.Delete(key) }
-func (z *ZkMerkleStateTrie) DeleteAccount(address common.Address) error       { return z.Delete(address[:]) }
+func (z *ZkMerkleStateTrie) DeleteStorage(_ common.Address, key []byte) error { return z.delete(key) }
+func (z *ZkMerkleStateTrie) DeleteAccount(address common.Address) error       { return z.delete(address[:]) }
+
+func (z *ZkMerkleStateTrie) delete(key []byte) error {
+	if err := z.Delete(key); errors.Is(err, trie.ErrKeyNotFound) {
+		return nil
```

Comment:
How about adding warn log that non-existing key is tried to deleted? with key in hex.